src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+
+/**
+ * The operations of task list.
+ */
+public class TaskList {
+    ArrayList<Task> tasks; //The tasklist.
+
+    /**
+     * Initiates the object with empty list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Initiates the object with tasklist.
+     *
+     * @param tasks The task list.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Deletes task from task list.
+     *
+     * @param n The order should be delete.
+     * @return The task be deleted.
+     * @throws DukeException When n > tasklist's size.
+     */
+    public Task deleteTask(int n) throws DukeException {
+        if (n > tasks.size()) {
```

Comment:
In the methods in the **TaskList** class, it may be good for you to check if the index is less than 1 in the two methods to make the check more comprehensive. Also, in doneTask(), maybe you could retrieve the task first and assign it to a variable, then mark it as done and return it? Because using 2 get() means we have to traverse the list twice which may add more complexity to the method.

Maybe you could also make a size() method too to return the size of the taskList instead of retrieving its array list and get its size. This way can make your code to adhere to the abstraction principle better.